mitodl/mitxpro
# Coupon functions and model changes

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #70 

#### What's this PR do?
- Adds functions for retrieving and validating coupons, applying a coupon to a bucket, redeeming a coupon, and determining the price for a product after a coupon is applied.
- Adds an `enabled` field to the `Coupon` model
- Adds an `automatic` boolean field to the `CouponInvoiceVersion` model

#### How should this be manually tested?
- Run the tests, they should pass
- If you feel ambitious, create a group of relevant coupon objects (see the `ecommerce.api_test.basket_and_coupons` fixture for examples) and try out the functions for them in the shell
